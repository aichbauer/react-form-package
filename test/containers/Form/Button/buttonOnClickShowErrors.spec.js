import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form (validate, validateOnClick) with button | click button |Â show errors', () => {
  const myComponent = (
    <Form
      validate
      validateOnClick
    >
      <div>
        <div>email</div>
        <Field type="email" id="email" required errorMessage="error" />
      </div>
      <div>
        <Button id="submit" type="submit" onClick={() => {}}>submit</Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);

  wrapper.find('button').simulate('click');
  wrapper.update();

  const errorMessage = wrapper.find('.rfp-error-label').text();

  expect(errorMessage).toBe('error');
  expect(tree).toMatchSnapshot();
});
