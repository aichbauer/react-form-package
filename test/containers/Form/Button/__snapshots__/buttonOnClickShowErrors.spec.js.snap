// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form (validate, validateOnClick) with button | click button |Â show errors 1`] = `
<form
  className="rfp-form"
>
  <div>
    <div>
      email
    </div>
    <input
      className="rfp-input"
      id="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      id="submit"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</form>
`;
